tecan_od_analyzer/__main__.py

```diff
@@ -697,26 +577,148 @@ def main():
                                     colnames[col]: clean_series.values,
                                 }
                             )
-                            gr_plots(
-                                df,
-                                colnames[col],
-                                color_=color_palette,
-                                legend_="exclude",
-                                title_="species",
-                                interpolationplot=interpolationplot,
-                                flag_svg=flag_svg,
-                            )
+
+                            # First time
+                            if start_leg == "":
+
+                                gr_plots(
+                                    df,
+                                    colnames[col],
+                                    color_=color_dict[bioshaker_label],
+                                    legend_="bioshaker",
+                                    title_="species",
+                                    interpolationplot=interpolationplot,
+                                    flag_svg=flag_svg,
+                                )
+                                start_leg = (colnames[col])[:3]
+
+                            # New Bioshaker
+                            elif (colnames[col])[:3] != start_leg:
+
+                                gr_plots(
+                                    df,
+                                    colnames[col],
+                                    color_=color_dict[bioshaker_label],
+                                    legend_="bioshaker",
+                                    title_="species",
+                                    interpolationplot=interpolationplot,
+                                    flag_svg=flag_svg,
+                                )
+                                start_leg = (colnames[col])[:3]
+
+                            # Repeated bioshaker
+                            else:
+                                gr_plots(
+                                    df,
+                                    colnames[col],
+                                    color_=color_dict[bioshaker_label],
+                                    legend_="exclude",
+                                    title_="species",
+                                    interpolationplot=interpolationplot,
+                                    flag_svg=flag_svg,
+                                )
 
                         last_name = colnames[col]
                         bioshaker_ = last_name[:3]
```

Comment:
Is this line and the one below with the regular expression what is causing the issues with naming and the plots?

Reply:
Yes that's the one